hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/impl/ChunkManagerImpl.java

```diff
@@ -210,19 +216,39 @@ public ByteBuffer readChunk(Container container, BlockID blockID,
     // of the chunk file.
     if (containerData.getLayOutVersion() == ChunkLayOutVersion
         .getLatestVersion().getVersion()) {
-      File chunkFile = ChunkUtils.getChunkFile(containerData, info);
 
-      // In case the chunk file does not exist but tmp chunk file exist,
-      // read from tmp chunk file if readFromTmpFile is set to true
-      if (!chunkFile.exists() && dispatcherContext != null
-          && dispatcherContext.isReadFromTmpFile()) {
-        chunkFile = getTmpChunkFile(chunkFile, dispatcherContext);
+      File finalChunkFile = ChunkUtils.getChunkFile(containerData, info);
+      File tmpChunkFile = getTmpChunkFile(finalChunkFile, dispatcherContext);
+
+      List<File> possibleFiles = new ArrayList<>();
+      possibleFiles.add(finalChunkFile);
+      if (dispatcherContext.isReadFromTmpFile()) {
```

Comment:
`null` check should be kept, or `TestDeleteWithSlowFollower` fixed:

https://github.com/apache/hadoop-ozone/blob/528cea3a7df6c96cb2ba633d4cd5ac35b1a6391b/hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestDeleteWithSlowFollower.java#L246-L248

Reply:
Fixed. Thanks.